charts/wildfly-common/templates/_ingress.yaml

```diff
@@ -0,0 +1,41 @@
+{{- define "wildfly-common.ingress.tpl" -}}
+kind: Ingress
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ include "wildfly-common.appName" . }}
+  labels: {}
+  annotations: {}
+spec:
+  {{- if .Values.deploy.ingress.className }}
+  ingressClassName: {{ .Values.deploy.ingress.className }}
+  {{- end }}
+  {{- if .Values.deploy.ingress.tls.enabled }}
```

Comment:
I wonder if we really need `deploy.ingress.tls.enabled`... we could verify if there is a `deploy.ingress.tls.secret` instead. There can not be a TLS config without that secret anyway.

From an user perspective, this is simpler:

```
deploy:
  tls:
    secret: my-tls-secret
```
@ehsavoie wdyt?